### DISCUSSION TITLE
Can't change my input data using x-model

### DISCUSSION THREAD

**Discussion by dticed** [alpinejs/alpine#4185] [1. Help]
Hi guys! I'm trying to implement a plugin for WordPress and I'm using Alpinejs to create the view for a meta-box.
My view:

```
<div x-data="erm()">
    <input id="related-movie" type="text" x-model="msg" @input="getType">
 
    <span x-text="msg"></span> | <span x-text="getType"></span>
</div>
```

My script:

```
document.addEventListener("alpine:init", () => {
  Alpine.data("erm", () => ({
    msg: "",
    
    init() {
        console.log(`init`);
      console.log(typeof this.msg);
    },

    getType() {
        return typeof this.msg;
    }
  }));
});
```

The issue:
Every time I try to insert data on my input field, it changes to type number and gets a 0 from somewhere. Can somebody help with this?

Demo:
https://github.com/alpinejs/alpine/assets/11221061/f0025877-cc1c-4800-944d-e5b0ad3d636f

**Comment by ekwoka** [comment]
[seems to work fine](https://awesomealpine.com/play/?coreplugins=0&script=QWxwaW5lLmRhdGEoJ2VybScsICgpID0-ICh7CiAgbXNnOiAnJywKCiAgaW5pdCgpIHsKICAgIGNvbnNvbGUubG9nKGBpbml0YCk7CiAgICBjb25zb2xlLmxvZyh0eXBlb2YgdGhpcy5tc2cpOwogIH0sCgogIGdldFR5cGUoKSB7CiAgICByZXR1cm4gdHlwZW9mIHRoaXMubXNnOwogIH0sCn0pKTsK&html=PGRpdgogIHgtZGF0YT0iZXJtKCkiCiAgc3R5bGU9ImNvbG9yOndoaXRlIgo-CiAgPGlucHV0CiAgICBpZD0icmVsYXRlZC1tb3ZpZSIKICAgIHR5cGU9InRleHQiCiAgICB4LW1vZGVsPSJtc2ciCiAgICBAaW5wdXQ9ImdldFR5cGUiCiAgICBzdHlsZT0iYmFja2dyb3VuZDogZ3JheTsgY29sb3I6IHdoaXRlIgogIC8-CgogIDxzcGFuIHgtdGV4dD0ibXNnIj48L3NwYW4-IHwKICA8c3BhbiB4LXRleHQ9ImdldFR5cGUiPjwvc3Bhbj4KPC9kaXY-Cg)

You have some other code doing something

**Reply by dticed** [reply]
Yes. It is probably some other script that is doing something...
I tried adding the `.lazy` to the x-model and it worked. Do you have any idea what this could be?
